src/index.ts

```diff
@@ -854,6 +870,63 @@ export abstract class PinejsClientCore<PinejsClient> {
 		}
 	}
 
+	private canRetryDefaultHandler(err: any) {
+		const code = err?.statusCode;
+		return code == null || code === 429 || (code >= 500 && code < 600);
+	}
+
+	protected async callWithRetry<T>(
+		fnCall: () => Promise<T>,
+		retry: RetryParameters,
+	): Promise<T> {
+		if (!retry || !this.retry) {
+			return await fnCall();
+		}
+
+		const minDelayMs = retry.minDelayMs ?? this.retry.minDelayMs ?? 0;
+		const maxDelayMs = retry.maxDelayMs ?? this.retry.maxDelayMs ?? 0;
+		const maxAttempts = retry.maxAttempts ?? this.retry.maxAttempts ?? 0;
+
+		if (minDelayMs <= 0) {
+			throw new Error('pinejs-client minDelayMs must be positive');
```

Comment:
I realized we can actually make this even more useful with
```suggestion
		if (minDelayMs == null || minDelayMs <= 0) {
			throw new Error(`pinejs-client minDelayMs must be a positive integer, got: '${minDelayMs}'`);
```
and no `?? 0` above. (I know it might seem overkill but the amount of time I've had saved by good error messages and lost by bad error messages is insane)

Reply:
good point!